core/commands/repo.go

```diff
@@ -95,3 +104,62 @@ order to reclaim hard disk space.
 		},
 	},
 }
+
+var repoStatCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline:          "Print status of the local repo.",
+		ShortDescription: ``,
+	},
+	Run: func(req cmds.Request, res cmds.Response) {
+		ctx := req.Context()
+		n, err := req.InvocContext().GetNode()
+		if err != nil {
+			res.SetError(err, cmds.ErrNormal)
+			return
+		}
+
+		usage, err := n.Repo.GetStorageUsage()
+		if err != nil {
+			res.SetError(err, cmds.ErrNormal)
+			return
+		}
+
+		allKeys, err := n.Blockstore.AllKeysChan(ctx)
+		if err != nil {
+			res.SetError(err, cmds.ErrNormal)
+			return
+		}
+
+		count := uint64(0)
+		for range allKeys {
```

Comment:
hrm... this is awkward. I think this is actually the best way of doing this, short of traversing the directory and listing all the files